commands/combat.py

```diff
@@ -545,6 +549,10 @@ def func(self):
                 drain_check(attack, attacker, caller, final_damage)
             record_combat(caller, action, "block", True, final_damage)
 
+        if "Long-Range" in attack.effects and attacker.key not in caller.db.ranged_knockback[1]:
```

Comment:
This block of logic can probably be tossed in a function since it's repeated in a few commands